src/Parsers/ExpressionListParsers.cpp

```diff
@@ -2120,6 +2141,55 @@ class ViewLayer : public Layer
     bool if_permitted;
 };
 
+/// Layer for table function 'kql'
+class KustoLayer : public Layer
+{
+public:
+
+    KustoLayer() : Layer(/*allow_alias*/ true, /*allow_alias_without_as_keyword*/ true) {}
+
+    bool parse(IParser::Pos & pos, Expected & expected, Action & /*action*/) override
+    {
+        /// kql(table|project ...)
+        /// 0. Parse the kql query
+        /// 1. Parse closing token
+        if (state == 0)
```

Comment:
I guess you forgot to change state in this `if`...

Reply:
right, fixed !